pkg/sidecar/k8s_reactor.go

```diff
@@ -37,26 +37,37 @@ var (
 )
 
 type K8sReactor struct {
-	client         *sync.Client
-	manager        *docker.Manager
-	servicesRoutes []net.IP
+	client          *sync.Client
+	manager         *docker.Manager
+	allowedServices []AllowedService
 }
 
 func NewK8sReactor() (Reactor, error) {
-	// TODO: Generalize this to a list of services.
-	wantedRoutes := []string{
-		os.Getenv(EnvRedisHost),
-		os.Getenv(EnvInfluxdbHost),
+	wantedServices := []struct {
```

Comment:
As a next improvement, I suggest we resolve the `wanted services` prior to allowing the routes for a given container, with a cache of 1-2min. or so.

This would allow the sidecar to recover relatively fast in case `redis` or `influxdb` or any other service is moved. WDYT?

Reply:
SGTM. We could also resolve once per run_id -- that removes the timing dependency.

That way, containers belonging to a given run_id scheduled on a given node would only resolve the services once. Once all containers are deleted, we would clean up the cache for that run_id.

Reply:
Sounds good. I will see about updating this in a future PR.